packages/grid/x-grid-data-generator/src/services/real-data-service.ts

```diff
@@ -12,8 +13,10 @@ export interface GridData {
 
 export function getRealData(rowLength: number, columns: any[]): Promise<GridData> {
   return new Promise<GridData>((resolve) => {
+    const tasks = { current: rowLength };
     const data: DataRowModel[] = [];
-    for (let i = 0; i < rowLength; i += 1) {
+
+    function work() {
```

Comment:
why not declare it ouside this scope?

Reply:
It needs to access the tasks. It can't be outside this scope